kythe/cxx/indexer/textproto/analyzer.cc

```diff
@@ -217,6 +236,120 @@ Status TextprotoAnalyzer::AnalyzeMessage(
   return OkStatus();
 }
 
+// Given a type url that looks like "type.googleapis.com/example.Message1",
+// returns "example.Message1".
+std::string ProtoMessageNameFromAnyTypeUrl(absl::string_view type_url) {
+  std::vector<std::string> parts = absl::StrSplit(type_url, "/");
+  if (parts.size() != 2) {
+    LOG(ERROR) << "Invalid Any type_url: '" << type_url << "'";
+    return "";
+  }
+  return parts[1];
+}
+
+// Example textproto:
+//   any_field {
+//     [some.url/mypackage.MyMessage] {
+//     }
+//   }
+//
+// Given the start location of "any_field" as field_loc, this function uses a
+// regex to find the "mypackage.MyMessage" portion and add an anchor node and a
+// ref to the proto message descriptor. Ideally this information would be
+// provided in the ParseInfoTree generated by the textproto parser, but since
+// it's not, we do our own "parsing" with a regex.
+Status TextprotoAnalyzer::AnalyzeAnyTypeUrl(
+    const proto::VName& file_vname, const Descriptor& msg_descriptor,
+    TextFormat::ParseLocation field_loc) {
+  // Note that line is 1-indexed; a value of zero indicates an empty location.
+  if (field_loc.line == 0) return OkStatus();
+
+  re2::StringPiece sp(textproto_content_.begin(), textproto_content_.size());
```

Comment:
should likely be `textproto_content_.data()`

Reply:
done